lib/ex_public_key.ex

```diff
@@ -314,10 +314,10 @@ defmodule ExPublicKey do
   end
 
   defp otp_has_rsa_gen_support do
-    case (to_string(Application.spec(:public_key, :vsn)) |> Float.parse) do
-      :error -> false
-      {version, _remainder} -> version >= 1.5
-    end
+    Application.spec(:public_key, :vsn)
+    |> Kernel.to_string
+    # 1.4.1 corresponds with OTP 20.0, which included crypto 4.0 that added the rsa public key generation. public_key the get an interface, too.
+    |> Version.match?(">= 1.4.1")
```

Comment:
I'm not sure using just the `Version` module from Elixir will work with OTP application versions. For example, the `public_key` version from OTP 18.2 is `1.1` which does not validate according to `Version`.

```
iex(10)> Version.parse("1.1")
:error
```

We probably can't use `Float.parse` either since some OTP applications carry versions like `1.4.1` as you've mentioned. The only options I can think of at the moment are to make the comparison by parsing the version string into a List of integers:

```
iex(15)> [1, 1] >= [1, 4, 1]
false
iex(16)> [1, 5] >= [1, 4, 1]
true
```

Or just checking the OTP release version:

```
iex(14)> :erlang.system_info(:otp_release)
'18'
```

Thoughts?

Reply:
The behaviour of `Version.parse` for non-patch versions surprises me, non-patch versions are common enough that I'd expected it to work - especially considering the documentation lists non-patch versions is some of the tables. If `Version.parse` returns an error, then a fallback to `Float.parse` might work to check if it's a float, and then prefix `".0"` to the string if it is indeed a Float. That should make the version parsing sane. I'll have a look to see what I can do with this to make it better.

I asked @barttenbrinke why he went with checking the `:vsn` of the erlang library, and he said that it's what Jose had told him (presuming he asked on IRC). My initial thought was simplifying to OTX release as well :-)

Reply:
I'm thinking that it's due to Elixir enforcing the [SemVer](http://semver.org/) spec and Erlang/OTP apps not using SemVer compliant version numbers. Technically, the missing patch version means the OTP app version number would be invalid under SemVer.

Since we are not comparing two SemVer version numbers, it seems like the simplest way to make the comparison work in call cases is to use the array-of-integers comparison method. That seems a lot more straight forward than trying to kludge the OTP app version into a SemVer in order to use Elixir's `Version` module to compare two non-SemVer version numbers.

I did a little searching to see if there was a canonical way of comparing OTP app version numbers but I didn't find anything. If there is a canonical way of doing it in Erlang we should use that method or just use the array comparison.

Reply:
Or, like you said, just check the OTP release :)